xbmc/utils/URIUtils.h

```diff
@@ -29,6 +29,15 @@ enum class LanCheckMode
 class URIUtils
 {
 public:
+  // Don't URL encode "-_.!()" according to RFC1738
+  //! @todo Update it to "-_.~" after Gotham according to RFC3986
+  static constexpr std::string_view RFC1738{"-._!()"};
+  static constexpr std::string_view RFC3986{"-_.~"};
+
+  static std::string URLEncode(std::string_view strURLData,
+                               const std::string_view& URLSpec = RFC1738);
```

Comment:
There is basically never a situation where passing a `std::string_view` by const reference is better than passing by value.
```suggestion
  static std::string URLEncode(std::string_view strURLData, std::string_view URLSpec = RFC1738);
```